include/evse_security/evse_security.hpp

```diff
@@ -189,8 +197,8 @@ class EvseSecurity {
 
 private:
     // Internal versions of the functions do not lock the mutex
-    InstallCertificateResult verify_certificate_internal(const std::string& certificate_chain,
-                                                         LeafCertificateType certificate_type);
+    CertificateValidationError verify_certificate_internal(const std::string& certificate_chain,
+                                                           CaCertificateType certificate_type);
```

Comment:
The problem with this approach of using 'CaCertificateType' is that there are some special cases that exist only in the 'LeafCertificateType' (there is a separate branch doing that) and not in 'CaCertificateType'. We should move back to 'LeafCertificateType' and add there the missing type.